app/models/exercise_prerequisite.rb

```diff
@@ -0,0 +1,9 @@
+class ExercisePrerequisite
+  include Neo4j::ActiveRel
+  include Neo4j::Timestamps
+
+  from_class :Exercise
+  to_class :Exercise
+
+  type 'prerequisite_of'
```

Comment:
Neo4j has a special kind of query to help model this kind of prerequisite tree.

```
(exercise:Exercise)-[:prerequisite_of*]->(next_exercise:Exercise)
return next_exercise
```

Because of the star, this will eagerly walk along a list of `[:prerequisite_of]` relationships and fetch all the exercises until you're at the end of the tree. You can also write therevers of this query:

```
(exercise:Exercise)<-[:prerequisite_of*]-(previous_exercise:Exercise)
return previous_exercise
```

The `>`, `<` hint to the graph which direction to walk.